homeassistant/components/opentherm_gw/__init__.py

```diff
@@ -419,9 +423,22 @@ async def cleanup(self, event=None):
     async def connect_and_subscribe(self):
         """Connect to serial device and subscribe report handler."""
         self.status = await self.gateway.connect(self.hass.loop, self.device_path)
-        _LOGGER.debug("Connected to OpenTherm Gateway at %s", self.device_path)
-        self.gw_version = self.status.get(gw_vars.OTGW_BUILD)
-
+        version_string = self.status[gw_vars.OTGW].get(gw_vars.OTGW_ABOUT)
+        self.gw_version = version_string[18:] if version_string else ""
+        _LOGGER.debug(
+            "Connected to OpenTherm Gateway %s at %s", self.gw_version, self.device_path
+        )
+        dev_reg = await async_get_dev_reg(self.hass)
+        gw_dev = dev_reg.async_get_or_create(
+            config_entry_id=self.config_entry_id,
+            identifiers={(DOMAIN, self.gw_id)},
+            name=self.name,
+            manufacturer="Schelte Bron",
+            model="OpenTherm Gateway",
+            sw_version=self.gw_version,
```

Comment:
If it's an empty string we shouldn't set the version at all.

Reply:
Thanks, fixed in 61445dd

Reply:
Proper fix in a36c22f